src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,250 @@
+#! python
+# -*- coding: utf-8 -*-
+# SkipGui.py
+# 2022, by Mark Ganson <TheMarkster>
+# LGPL 2.1 or later
+
+# this file is called from c++ in Mod/Part/Gui/Command.cpp
+# from various functions when these commands are activated:
+#   Part::Loft
+#   Part::Sweep
+#   Part::Extrude
+#   Part::Revolve
+# Then if the user has preselected certain objects and subobjects
+# the Gui can be skipped and the document objects directly created
+
+# If False is returned, then the Gui dialog should be shown
+# Else if True is returned, then the dialog can be skipped
+
+import FreeCAD as App
+import FreeCADGui as Gui
+
+def getActStr(document):
+    return "App.getDocument(\'" + document.Name + "\').ActiveObject"
+
+def getDocStr(document):
+    return "App.getDocument(\'" + document.Name +"\')"
+
+def getObjStr(document, object):
+    return getDocStr(document) + ".getObject(\'" + object.Name + "\')"
+
+def getVisStr(document, objs):
+    ret = ""
+    for obj in objs:
+        ret += getDocStr(document) + ".getObject(\'" + obj.Name + "\').ViewObject.Visibility = False\n"
```

Comment:
Formatted string. No single quote escape needed.

Name of the function is a bit weird as it eventually not only get things, but set Visibility to False.

Reply:
All of these begin with get because they are getting a string value, here the string it gets is the string used to set visibility to false.  I think of it has getVisibilityStringForSettingVisibilityToFalse(), but this was less typing.